Isosurfacing_3/include/CGAL/Isosurfacing_3/internal/Dual_contouring_internal.h

```diff
@@ -0,0 +1,362 @@
+    /// <param name="k"></param>
+    /// <returns> true, if there is a point in the cell</returns>
+    template <class Domain_>
+    bool position(const Domain_& domain, const typename Domain_::FT iso_value, const typename Domain_::Cell_handle& vh,
+                  typename Domain_::Point& point) const {
+        typedef typename Domain_::Point Point;
+        typedef typename Domain_::Geom_traits::Vector_3 Vector;
+        typedef typename Domain_::FT FT;
+
+        typename Domain_::Cell_vertices vertices = domain.cell_vertices(vh);
+
+        namespace Tables = internal::Cube_table;
+
+        std::array<typename Domain_::FT, Tables::N_VERTICES> s;
+        std::transform(vertices.begin(), vertices.end(), s.begin(), [&](const auto& v) { return domain.value(v); });
+
+        std::array<bool, Tables::N_VERTICES> b;
+        std::transform(s.begin(), s.end(), b.begin(), [iso_value](const auto& e) { return e <= iso_value; });
+
+        unsigned int cubeindex = 0;
+        // set bit if corresponding corner is below iso
+        for (int i = 0; i < Tables::N_VERTICES; ++i) {
+            cubeindex |= b[i] << i;
+        }
+
+        if (cubeindex == 0 || cubeindex == 255) {
+            return false;
+        }
+
+        std::array<Vector, Tables::N_VERTICES> pos;
+        std::transform(vertices.begin(), vertices.end(), pos.begin(),
+                       [&](const auto& v) { return domain.position(v) - CGAL::ORIGIN; });
+
+        point = CGAL::ORIGIN + (pos[0] + 0.5 * (pos[7] - pos[0]));  // set point to voxel center
+
+        // std::array<Vector, Tables::N_VERTICES> normals;
+        // std::transform(vertices.begin(), vertices.end(), normals.begin(),
+        //                [&](const auto& v) { return domain.gradient(domain.position(v)); });
+
+        // compute edge intersections
+        std::vector<Point> edge_intersections;
+        std::vector<Vector> edge_intersection_normals;
+
+        for (int i = 0; i < Tables::N_EDGES; ++i) {
+            const auto& v0 = Tables::edge_to_vertex[i][0];
+            const auto& v1 = Tables::edge_to_vertex[i][1];
+
+            if (b[v0] != b[v1]) {  // e0
+                const FT u = (s[v0] - iso_value) / (s[v0] - s[v1]);
+                const Point p_lerp = CGAL::ORIGIN + ((1 - u) * pos[v0] + u * pos[v1]);
+                edge_intersections.push_back(p_lerp);
+                // const Vector n_lerp = (1 - u) * normals[v0] + u * normals[v1];
+                edge_intersection_normals.push_back(domain.gradient(p_lerp));
+            }
+        }
+
+        // MC Polygon Center of Mass
+        if (false) {
+            Vector com_vec(0, 0, 0);
+
+            for (int i = 0; i < edge_intersections.size(); ++i) {
+                com_vec += edge_intersections[i] - CGAL::ORIGIN;
+            }
+
+            Point p = CGAL::ORIGIN + com_vec / edge_intersections.size();
+            point = p;
+        }
+
+        // SVD QEM
+        if (true) {
+            Eigen::Matrix3d A;
```

Comment:
Instead of using Eigen directly we usually have an abstraction layer.  See [this pacakge](https://doc.cgal.org/latest/Manual/packages.html#PkgSolverInterface)